packages/cloud_firestore/cloud_firestore/lib/src/query.dart

```diff
@@ -6,106 +6,45 @@ part of cloud_firestore;
-    return Query._(
-      firestore: firestore,
-      isCollectionGroup: _isCollectionGroup,
-      pathComponents: _pathComponents,
-      parameters: Map<String, dynamic>.unmodifiable(
-        Map<String, dynamic>.from(_parameters)..addAll(parameters),
-      ),
-    );
-  }
+  String get _path => _pathComponents.join('/');
 
   Map<String, dynamic> buildArguments() {
-    return Map<String, dynamic>.from(_parameters)
-      ..addAll(<String, dynamic>{
-        'path': _path,
-      });
+    return _delegate.buildArguments();
   }
 
   /// Notifies of query results at this location
-  // TODO(jackson): Reduce code duplication with [DocumentReference]
-  Stream<QuerySnapshot> snapshots({bool includeMetadataChanges = false}) {
-    assert(includeMetadataChanges != null);
-    Future<int> _handle;
-    // It's fine to let the StreamController be garbage collected once all the
-    // subscribers have cancelled; this analyzer warning is safe to ignore.
-    StreamController<QuerySnapshot> controller; // ignore: close_sinks
-    controller = StreamController<QuerySnapshot>.broadcast(
-      onListen: () {
-        _handle = Firestore.channel.invokeMethod<int>(
-          'Query#addSnapshotListener',
-          <String, dynamic>{
-            'app': firestore.app.name,
-            'path': _path,
-            'isCollectionGroup': _isCollectionGroup,
-            'parameters': _parameters,
-            'includeMetadataChanges': includeMetadataChanges,
-          },
-        ).then<int>((dynamic result) => result);
-        _handle.then((int handle) {
-          Firestore._queryObservers[handle] = controller;
-        });
-      },
-      onCancel: () {
-        _handle.then((int handle) async {
-          await Firestore.channel.invokeMethod<void>(
-            'removeListener',
-            <String, dynamic>{'handle': handle},
-          );
-          Firestore._queryObservers.remove(handle);
-        });
-      },
-    );
-    return controller.stream;
-  }
+  Stream<QuerySnapshot> snapshots({bool includeMetadataChanges = false}) =>
+      _delegate
+          .snapshots(includeMetadataChanges: includeMetadataChanges)
+          .map((item) => QuerySnapshot._(item));
 
   /// Fetch the documents for this query
   Future<QuerySnapshot> getDocuments(
       {Source source = Source.serverAndCache}) async {
     assert(source != null);
-    final Map<dynamic, dynamic> data =
-        await Firestore.channel.invokeMapMethod<String, dynamic>(
-      'Query#getDocuments',
-      <String, dynamic>{
-        'app': firestore.app.name,
-        'path': _path,
-        'isCollectionGroup': _isCollectionGroup,
-        'parameters': _parameters,
-        'source': _getSourceString(source),
-      },
-    );
-    return QuerySnapshot._(data, firestore);
+    try{
+      final docs = await _delegate.getDocuments(source: PlatformUtils._toPlatformSource(source));
+
+      return QuerySnapshot._(docs);
+    }catch(error) {
+
+      throw error;
+    }
```

Comment:
I think you forgot to remove this try/catch from some debugging?

In case you want to do something with `error`, you should use `rethrow;` instead. See [docs](https://dart.dev/guides/language/effective-dart/usage#do-use-rethrow-to-rethrow-a-caught-exception)